src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -308,6 +308,24 @@ private TestClassInfo CreateClassInfo(Type classType, TestMethod testMethod)
                         // Update test initialize/cleanup method from base type.
                         this.UpdateInfoIfTestInitializeOrCleanupMethod(classInfo, methodInfo, true, instanceMethods, testInitializeAttributeType, testCleanupAttributeType);
                     }
+
+                    if (this.IsAssemblyOrClassInitializeMethod(methodInfo, classInitializeAttributeType))
+                    {
+                        if (methodInfo.GetCustomAttribute<ClassInitializeAttribute>().InheritanceBehavior == ClassInitializeInheritance.None)
+                        {
+                            // if the ClassInitializeInheritance is None, means
+                            // it will pick only the derived class initalize method
+                            break;
```

Comment:
hmm, we probably might not want to break here. This would effect the TestInitialize methodinfo gathering above.

Reply:
hmm, right it could be. Changed to continue instead which i think make more sense.
make sense?

Reply:
I imagine you'd set a flag to not process any more Class Initializes/Class Cleanups within that loop.

Reply:
This was affected as part of the changes mentioned in below comment. 
So, instead of checking if the attribute is None, changed to see if the attribute is BeforeEachDerivedClass, then added the base method to the queue. Otherwise, nothing will be done and TestInitialize methodinfo gathering won't be affected at all by breaking or anything.

I'll push this changes after other review comments are resolved.